### PR TITLE
feature: modifying local configuration information is not allowed.

### PR DESCRIPTION
**Pull request by membphis**

### CODE REVIEW

File: lua/apisix/core/table.lua [hunk 1]
```diff
@@ -81,4 +82,18 @@ end
 _M.deepcopy = deepcopy
 
 
+function _M.readonly(t)
```
Comment (moonming) [inline lua/apisix/core/table.lua]: This function prohibits the insertion of new elements, but should it still be possible to modify existing data?

Comment (membphis) [reply lua/apisix/core/table.lua]: > should it still be possible to modify existing data?

Modification of existing data is not allowed. I'll add new test case to confirm this.

Comment (membphis) [reply lua/apisix/core/table.lua]: added more test case: https://github.com/apache/incubator-apisix/pull/1143/commits/9d0ffbe6cefe7dbd0ba70ae45baab007c89a39e5#diff-7e4b33723fb43f3a64bd46cb434958e7R67

**Comment by membphis** [comment]
It looks a bit complicated, so close it first.

**Pull request closed without merging**